optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-client/src/main/java/org/optaplanner/workbench/screens/solver/client/editor/TerminationTreeItemContentViewImpl.java

```diff
@@ -0,0 +1,613 @@
+        li2.addClickHandler( h -> {
+            presenter.onTerminationTypeSelected( TerminationConfigOption.UNIMPROVED_MILLISECONDS_SPENT_LIMIT.name() );
+        } );
+        dropdownMenuList.add( li2 );
+
+        dropDownNameMap.put( TerminationConfigOption.BEST_SCORE_LIMIT, I18N_CONSTANTS.BestScoreLimit() );
+        AnchorListItem li4 = new AnchorListItem( I18N_CONSTANTS.BestScoreLimit() );
+        li4.addClickHandler( h -> {
+            presenter.onTerminationTypeSelected( TerminationConfigOption.BEST_SCORE_LIMIT.name() );
+        } );
+        dropdownMenuList.add( li4 );
+
+        dropDownNameMap.put( TerminationConfigOption.BEST_SCORE_FEASIBLE, I18N_CONSTANTS.BestScoreFeasible() );
+        AnchorListItem li3 = new AnchorListItem( I18N_CONSTANTS.BestScoreFeasible() );
+        li3.addClickHandler( h -> {
+            presenter.onTerminationTypeSelected( TerminationConfigOption.BEST_SCORE_FEASIBLE.name() );
+        } );
+        dropdownMenuList.add( li3 );
+
+        dropDownNameMap.put( TerminationConfigOption.STEP_COUNT_LIMIT, I18N_CONSTANTS.StepCountLimit() );
+        AnchorListItem li6 = new AnchorListItem( I18N_CONSTANTS.StepCountLimit() );
+        li6.addClickHandler( h -> {
+            presenter.onTerminationTypeSelected( TerminationConfigOption.STEP_COUNT_LIMIT.name() );
+        } );
+        dropdownMenuList.add( li6 );
+
+        dropDownNameMap.put( TerminationConfigOption.UNIMPROVED_STEP_COUNT_LIMIT, I18N_CONSTANTS.UnimprovedStepCountLimit() );
+        AnchorListItem li7 = new AnchorListItem( I18N_CONSTANTS.UnimprovedStepCountLimit() );
+        li7.addClickHandler( h -> {
+            presenter.onTerminationTypeSelected( TerminationConfigOption.UNIMPROVED_STEP_COUNT_LIMIT.name() );
+        } );
+        dropdownMenuList.add( li7 );
+
+        dropDownNameMap.put( TerminationConfigOption.SCORE_CALCULATION_COUNT_LIMIT, I18N_CONSTANTS.ScoreCalculationCountLimit() );
+        AnchorListItem li5 = new AnchorListItem( I18N_CONSTANTS.ScoreCalculationCountLimit() );
+        li5.addClickHandler( h -> {
+            presenter.onTerminationTypeSelected( TerminationConfigOption.SCORE_CALCULATION_COUNT_LIMIT.name() );
+        } );
+        dropdownMenuList.add( li5 );
+
+        dropDownNameMap.put( TerminationConfigOption.NESTED, I18N_CONSTANTS.NestedTermination() );
+        AnchorListItem li8 = new AnchorListItem( I18N_CONSTANTS.NestedTermination() );
+        li8.addClickHandler( h -> {
+            presenter.onTerminationTypeSelected( TerminationConfigOption.NESTED.name() );
+        } );
+        dropdownMenuList.add( li8 );
+    }
+
+    @EventHandler("daysSpentInput")
+    public void handleDaysSpentInputChange( ChangeEvent event ) {
+        try {
+            Long value = daysSpentInput.getValue().isEmpty() ? null : Long.parseLong( daysSpentInput.getValue() );
+            presenter.onDaysSpentChange( value );
+        } catch ( NumberFormatException e ) {
+
```

Comment:
So this would leave the View in an inconsistent state with the Presenter? Should you not show an error in the View that the value was wrong? Or default to a valid value and update the View? etc